src/genai/services/async_generator.py

```diff
@@ -194,7 +194,7 @@ def _request_launcher(self):
         except Exception as ex:
             self.throw_on_error = True
             self._queue.put_nowait((1, 1, None, ex))
-            raise ex
+            raise to_genai_error(ex)
```

Comment:
Could you keep it as is? We want the same error in the queue (line above) to be the same as the error we just raised.